coding/wml-v20/src/main/java/org/n52/sos/aquarius/OmObservationMapper.java

```diff
@@ -97,4 +87,20 @@ public String toInterpolationTypeHref(final OmObservation observation) {
         return wml2InterpolationHref;
     }
 
+    private String mapProcedureIdentifierToInterpolationType(String procedureIdentifier) {
+        String aquariusInterpolationType = null;
+        if (null != procedureIdentifier && !procedureIdentifier.isEmpty()) {
+            int indexOfSeparationCharacter = procedureIdentifier.lastIndexOf(SEPARATOR_CHARACTER);
+            if (indexOfSeparationCharacter > 0) {
+                aquariusInterpolationType = procedureIdentifier.substring(indexOfSeparationCharacter + 1);
```

Comment:
Another strange edge case here, if you want to consider it. if the id ends with a "_", then we get an OutOfBounds here.